File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/ItemCounts.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
/**
 * Class LibrarySummary  
 */
@Schema(description = "Class LibrarySummary  ")

public class ItemCounts {
  @SerializedName("MovieCount")
  private Integer movieCount = null;

  @SerializedName("SeriesCount")
  private Integer seriesCount = null;

  @SerializedName("EpisodeCount")
  private Integer episodeCount = null;

  @SerializedName("GameCount")
  private Integer gameCount = null;

  @SerializedName("ArtistCount")
  private Integer artistCount = null;

  @SerializedName("ProgramCount")
  private Integer programCount = null;

  @SerializedName("GameSystemCount")
  private Integer gameSystemCount = null;

  @SerializedName("TrailerCount")
  private Integer trailerCount = null;

  @SerializedName("SongCount")
  private Integer songCount = null;

  @SerializedName("AlbumCount")
  private Integer albumCount = null;

  @SerializedName("MusicVideoCount")
  private Integer musicVideoCount = null;

  @SerializedName("BoxSetCount")
  private Integer boxSetCount = null;

  @SerializedName("BookCount")
  private Integer bookCount = null;

  @SerializedName("ItemCount")
  private Integer itemCount = null;

  public ItemCounts movieCount(Integer movieCount) {
    this.movieCount = movieCount;
    return this;
  }

   /**
   * The movie count.
   * @return movieCount
  **/
  @Schema(description = "The movie count.")
  public Integer getMovieCount() {
    return movieCount;
  }

  public void setMovieCount(Integer movieCount) {
    this.movieCount = movieCount;
  }

  public ItemCounts seriesCount(Integer seriesCount) {
    this.seriesCount = seriesCount;
    return this;
  }

   /**
   * The series count.
   * @return seriesCount
  **/
  @Schema(description = "The series count.")
  public Integer getSeriesCount() {
    return seriesCount;
  }

  public void setSeriesCount(Integer seriesCount) {
    this.seriesCount = seriesCount;
  }

  public ItemCounts episodeCount(Integer episodeCount) {
    this.episodeCount = episodeCount;
    return this;
  }

   /**
   * The episode count.
   * @return episodeCount
  **/
  @Schema(description = "The episode count.")
  public Integer getEpisodeCount() {
    return episodeCount;
  }

  public void setEpisodeCount(Integer episodeCount) {
    this.episodeCount = episodeCount;
  }

  public ItemCounts gameCount(Integer gameCount) {
    this.gameCount = gameCount;
    return this;
  }

   /**
   * The game count.
   * @return gameCount
  **/
  @Schema(description = "The game count.")
  public Integer getGameCount() {
    return gameCount;
  }

  public void setGameCount(Integer gameCount) {
    this.gameCount = gameCount;
  }

  public ItemCounts artistCount(Integer artistCount) {
    this.artistCount = artistCount;
    return this;
  }

   /**
   * Get artistCount
   * @return artistCount
  **/
  @Schema(description = "")
  public Integer getArtistCount() {
    return artistCount;
  }

  public void setArtistCount(Integer artistCount) {
    this.artistCount = artistCount;
  }

  public ItemCounts programCount(Integer programCount) {
    this.programCount = programCount;
    return this;
  }

   /**
   * Get programCount
   * @return programCount
  **/
  @Schema(description = "")
  public Integer getProgramCount() {
    return programCount;
  }

  public void setProgramCount(Integer programCount) {
    this.programCount = programCount;
  }

  public ItemCounts gameSystemCount(Integer gameSystemCount) {
    this.gameSystemCount = gameSystemCount;
    return this;
  }

   /**
   * The game system count.
   * @return gameSystemCount
  **/
  @Schema(description = "The game system count.")
  public Integer getGameSystemCount() {
    return gameSystemCount;
  }

  public void setGameSystemCount(Integer gameSystemCount) {
    this.gameSystemCount = gameSystemCount;
  }

  public ItemCounts trailerCount(Integer trailerCount) {
    this.trailerCount = trailerCount;
    return this;
  }

   /**
   * The trailer count.
   * @return trailerCount
  **/
  @Schema(description = "The trailer count.")
  public Integer getTrailerCount() {
    return trailerCount;
  }

  public void setTrailerCount(Integer trailerCount) {
    this.trailerCount = trailerCount;
  }

  public ItemCounts songCount(Integer songCount) {
    this.songCount = songCount;
    return this;
  }

   /**
   * The song count.
   * @return songCount
  **/
  @Schema(description = "The song count.")
  public Integer getSongCount() {
    return songCount;
  }

  public void setSongCount(Integer songCount) {
    this.songCount = songCount;
  }

  public ItemCounts albumCount(Integer albumCount) {
    this.albumCount = albumCount;
    return this;
  }

   /**
   * The album count.
   * @return albumCount
  **/
  @Schema(description = "The album count.")
  public Integer getAlbumCount() {
    return albumCount;
  }

  public void setAlbumCount(Integer albumCount) {
    this.albumCount = albumCount;
  }

  public ItemCounts musicVideoCount(Integer musicVideoCount) {
    this.musicVideoCount = musicVideoCount;
    return this;
  }

   /**
   * The music video count.
   * @return musicVideoCount
  **/
  @Schema(description = "The music video count.")
  public Integer getMusicVideoCount() {
    return musicVideoCount;
  }

  public void setMusicVideoCount(Integer musicVideoCount) {
    this.musicVideoCount = musicVideoCount;
  }

  public ItemCounts boxSetCount(Integer boxSetCount) {
    this.boxSetCount = boxSetCount;
    return this;
  }

   /**
   * The box set count.
   * @return boxSetCount
  **/
  @Schema(description = "The box set count.")
  public Integer getBoxSetCount() {
    return boxSetCount;
  }

  public void setBoxSetCount(Integer boxSetCount) {
    this.boxSetCount = boxSetCount;
  }

  public ItemCounts bookCount(Integer bookCount) {
    this.bookCount = bookCount;
    return this;
  }

   /**
   * The book count.
   * @return bookCount
  **/
  @Schema(description = "The book count.")
  public Integer getBookCount() {
    return bookCount;
  }

  public void setBookCount(Integer bookCount) {
    this.bookCount = bookCount;
  }

  public ItemCounts itemCount(Integer itemCount) {
    this.itemCount = itemCount;
    return this;
  }

   /**
   * Get itemCount
   * @return itemCount
  **/
  @Schema(description = "")
  public Integer getItemCount() {
    return itemCount;
  }

  public void setItemCount(Integer itemCount) {
    this.itemCount = itemCount;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ItemCounts itemCounts = (ItemCounts) o;
    return Objects.equals(this.movieCount, itemCounts.movieCount) &&
        Objects.equals(this.seriesCount, itemCounts.seriesCount) &&
        Objects.equals(this.episodeCount, itemCounts.episodeCount) &&
        Objects.equals(this.gameCount, itemCounts.gameCount) &&
        Objects.equals(this.artistCount, itemCounts.artistCount) &&
        Objects.equals(this.programCount, itemCounts.programCount) &&
        Objects.equals(this.gameSystemCount, itemCounts.gameSystemCount) &&
        Objects.equals(this.trailerCount, itemCounts.trailerCount) &&
        Objects.equals(this.songCount, itemCounts.songCount) &&
        Objects.equals(this.albumCount, itemCounts.albumCount) &&
        Objects.equals(this.musicVideoCount, itemCounts.musicVideoCount) &&
        Objects.equals(this.boxSetCount, itemCounts.boxSetCount) &&
        Objects.equals(this.bookCount, itemCounts.bookCount) &&
        Objects.equals(this.itemCount, itemCounts.itemCount);
  }

  @Override
  public int hashCode() {
    return Objects.hash(movieCount, seriesCount, episodeCount, gameCount, artistCount, programCount, gameSystemCount, trailerCount, songCount, albumCount, musicVideoCount, boxSetCount, bookCount, itemCount);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ItemCounts {\n");
    
    sb.append("    movieCount: ").append(toIndentedString(movieCount)).append("\n");
    sb.append("    seriesCount: ").append(toIndentedString(seriesCount)).append("\n");
    sb.append("    episodeCount: ").append(toIndentedString(episodeCount)).append("\n");
    sb.append("    gameCount: ").append(toIndentedString(gameCount)).append("\n");
    sb.append("    artistCount: ").append(toIndentedString(artistCount)).append("\n");
    sb.append("    programCount: ").append(toIndentedString(programCount)).append("\n");
    sb.append("    gameSystemCount: ").append(toIndentedString(gameSystemCount)).append("\n");
    sb.append("    trailerCount: ").append(toIndentedString(trailerCount)).append("\n");
    sb.append("    songCount: ").append(toIndentedString(songCount)).append("\n");
    sb.append("    albumCount: ").append(toIndentedString(albumCount)).append("\n");
    sb.append("    musicVideoCount: ").append(toIndentedString(musicVideoCount)).append("\n");
    sb.append("    boxSetCount: ").append(toIndentedString(boxSetCount)).append("\n");
    sb.append("    bookCount: ").append(toIndentedString(bookCount)).append("\n");
    sb.append("    itemCount: ").append(toIndentedString(itemCount)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: SampleCode/RestApi/Clients/Net.RestSharp/EmbyClient.Dotnet/Model/ItemCounts.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Class LibrarySummary  
    /// </summary>
    [DataContract]
        public partial class ItemCounts :  IEquatable<ItemCounts>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ItemCounts" /> class.
        /// </summary>
        /// <param name="movieCount">The movie count..</param>
        /// <param name="seriesCount">The series count..</param>
        /// <param name="episodeCount">The episode count..</param>
        /// <param name="gameCount">The game count..</param>
        /// <param name="artistCount">artistCount.</param>
        /// <param name="programCount">programCount.</param>
        /// <param name="gameSystemCount">The game system count..</param>
        /// <param name="trailerCount">The trailer count..</param>
        /// <param name="songCount">The song count..</param>
        /// <param name="albumCount">The album count..</param>
        /// <param name="musicVideoCount">The music video count..</param>
        /// <param name="boxSetCount">The box set count..</param>
        /// <param name="bookCount">The book count..</param>
        /// <param name="itemCount">itemCount.</param>
        public ItemCounts(int? movieCount = default(int?), int? seriesCount = default(int?), int? episodeCount = default(int?), int? gameCount = default(int?), int? artistCount = default(int?), int? programCount = default(int?), int? gameSystemCount = default(int?), int? trailerCount = default(int?), int? songCount = default(int?), int? albumCount = default(int?), int? musicVideoCount = default(int?), int? boxSetCount = default(int?), int? bookCount = default(int?), int? itemCount = default(int?))
        {
            this.MovieCount = movieCount;
            this.SeriesCount = seriesCount;
            this.EpisodeCount = episodeCount;
            this.GameCount = gameCount;
            this.ArtistCount = artistCount;
            this.ProgramCount = programCount;
            this.GameSystemCount = gameSystemCount;
            this.TrailerCount = trailerCount;
            this.SongCount = songCount;
            this.AlbumCount = albumCount;
            this.MusicVideoCount = musicVideoCount;
            this.BoxSetCount = boxSetCount;
            this.BookCount = bookCount;
            this.ItemCount = itemCount;
        }
        
        /// <summary>
        /// The movie count.
        /// </summary>
        /// <value>The movie count.</value>
        [DataMember(Name="MovieCount", EmitDefaultValue=false)]
        public int? MovieCount { get; set; }

        /// <summary>
        /// The series count.
        /// </summary>
        /// <value>The series count.</value>
        [DataMember(Name="SeriesCount", EmitDefaultValue=false)]
        public int? SeriesCount { get; set; }

        /// <summary>
        /// The episode count.
        /// </summary>
        /// <value>The episode count.</value>
        [DataMember(Name="EpisodeCount", EmitDefaultValue=false)]
        public int? EpisodeCount { get; set; }

        /// <summary>
        /// The game count.
        /// </summary>
        /// <value>The game count.</value>
        [DataMember(Name="GameCount", EmitDefaultValue=false)]
        public int? GameCount { get; set; }

        /// <summary>
        /// Gets or Sets ArtistCount
        /// </summary>
        [DataMember(Name="ArtistCount", EmitDefaultValue=false)]
        public int? ArtistCount { get; set; }

        /// <summary>
        /// Gets or Sets ProgramCount
        /// </summary>
        [DataMember(Name="ProgramCount", EmitDefaultValue=false)]
        public int? ProgramCount { get; set; }

        /// <summary>
        /// The game system count.
        /// </summary>
        /// <value>The game system count.</value>
        [DataMember(Name="GameSystemCount", EmitDefaultValue=false)]
        public int? GameSystemCount { get; set; }

        /// <summary>
        /// The trailer count.
        /// </summary>
        /// <value>The trailer count.</value>
        [DataMember(Name="TrailerCount", EmitDefaultValue=false)]
        public int? TrailerCount { get; set; }

        /// <summary>
        /// The song count.
        /// </summary>
        /// <value>The song count.</value>
        [DataMember(Name="SongCount", EmitDefaultValue=false)]
        public int? SongCount { get; set; }

        /// <summary>
        /// The album count.
        /// </summary>
        /// <value>The album count.</value>
        [DataMember(Name="AlbumCount", EmitDefaultValue=false)]
        public int? AlbumCount { get; set; }

        /// <summary>
        /// The music video count.
        /// </summary>
        /// <value>The music video count.</value>
        [DataMember(Name="MusicVideoCount", EmitDefaultValue=false)]
        public int? MusicVideoCount { get; set; }

        /// <summary>
        /// The box set count.
        /// </summary>
        /// <value>The box set count.</value>
        [DataMember(Name="BoxSetCount", EmitDefaultValue=false)]
        public int? BoxSetCount { get; set; }

        /// <summary>
        /// The book count.
        /// </summary>
        /// <value>The book count.</value>
        [DataMember(Name="BookCount", EmitDefaultValue=false)]
        public int? BookCount { get; set; }

        /// <summary>
        /// Gets or Sets ItemCount
        /// </summary>
        [DataMember(Name="ItemCount", EmitDefaultValue=false)]
        public int? ItemCount { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ItemCounts {\n");
            sb.Append("  MovieCount: ").Append(MovieCount).Append("\n");
            sb.Append("  SeriesCount: ").Append(SeriesCount).Append("\n");
            sb.Append("  EpisodeCount: ").Append(EpisodeCount).Append("\n");
            sb.Append("  GameCount: ").Append(GameCount).Append("\n");
            sb.Append("  ArtistCount: ").Append(ArtistCount).Append("\n");
            sb.Append("  ProgramCount: ").Append(ProgramCount).Append("\n");
            sb.Append("  GameSystemCount: ").Append(GameSystemCount).Append("\n");
            sb.Append("  TrailerCount: ").Append(TrailerCount).Append("\n");
            sb.Append("  SongCount: ").Append(SongCount).Append("\n");
            sb.Append("  AlbumCount: ").Append(AlbumCount).Append("\n");
            sb.Append("  MusicVideoCount: ").Append(MusicVideoCount).Append("\n");
            sb.Append("  BoxSetCount: ").Append(BoxSetCount).Append("\n");
            sb.Append("  BookCount: ").Append(BookCount).Append("\n");
            sb.Append("  ItemCount: ").Append(ItemCount).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as ItemCounts);
        }

        /// <summary>
        /// Returns true if ItemCounts instances are equal
        /// </summary>
        /// <param name="input">Instance of ItemCounts to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ItemCounts input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.MovieCount == input.MovieCount ||
                    (this.MovieCount != null &&
                    this.MovieCount.Equals(input.MovieCount))
                ) && 
                (
                    this.SeriesCount == input.SeriesCount ||
                    (this.SeriesCount != null &&
                    this.SeriesCount.Equals(input.SeriesCount))
                ) && 
                (
                    this.EpisodeCount == input.EpisodeCount ||
                    (this.EpisodeCount != null &&
                    this.EpisodeCount.Equals(input.EpisodeCount))
                ) && 
                (
                    this.GameCount == input.GameCount ||
                    (this.GameCount != null &&
                    this.GameCount.Equals(input.GameCount))
                ) && 
                (
                    this.ArtistCount == input.ArtistCount ||
                    (this.ArtistCount != null &&
                    this.ArtistCount.Equals(input.ArtistCount))
                ) && 
                (
                    this.ProgramCount == input.ProgramCount ||
                    (this.ProgramCount != null &&
                    this.ProgramCount.Equals(input.ProgramCount))
                ) && 
                (
                    this.GameSystemCount == input.GameSystemCount ||
                    (this.GameSystemCount != null &&
                    this.GameSystemCount.Equals(input.GameSystemCount))
                ) && 
                (
                    this.TrailerCount == input.TrailerCount ||
                    (this.TrailerCount != null &&
                    this.TrailerCount.Equals(input.TrailerCount))
                ) && 
                (
                    this.SongCount == input.SongCount ||
                    (this.SongCount != null &&
                    this.SongCount.Equals(input.SongCount))
                ) && 
                (
                    this.AlbumCount == input.AlbumCount ||
                    (this.AlbumCount != null &&
                    this.AlbumCount.Equals(input.AlbumCount))
                ) && 
                (
                    this.MusicVideoCount == input.MusicVideoCount ||
                    (this.MusicVideoCount != null &&
                    this.MusicVideoCount.Equals(input.MusicVideoCount))
                ) && 
                (
                    this.BoxSetCount == input.BoxSetCount ||
                    (this.BoxSetCount != null &&
                    this.BoxSetCount.Equals(input.BoxSetCount))
                ) && 
                (
                    this.BookCount == input.BookCount ||
                    (this.BookCount != null &&
                    this.BookCount.Equals(input.BookCount))
                ) && 
                (
                    this.ItemCount == input.ItemCount ||
                    (this.ItemCount != null &&
                    this.ItemCount.Equals(input.ItemCount))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.MovieCount != null)
                    hashCode = hashCode * 59 + this.MovieCount.GetHashCode();
                if (this.SeriesCount != null)
                    hashCode = hashCode * 59 + this.SeriesCount.GetHashCode();
                if (this.EpisodeCount != null)
                    hashCode = hashCode * 59 + this.EpisodeCount.GetHashCode();
                if (this.GameCount != null)
                    hashCode = hashCode * 59 + this.GameCount.GetHashCode();
                if (this.ArtistCount != null)
                    hashCode = hashCode * 59 + this.ArtistCount.GetHashCode();
                if (this.ProgramCount != null)
                    hashCode = hashCode * 59 + this.ProgramCount.GetHashCode();
                if (this.GameSystemCount != null)
                    hashCode = hashCode * 59 + this.GameSystemCount.GetHashCode();
                if (this.TrailerCount != null)
                    hashCode = hashCode * 59 + this.TrailerCount.GetHashCode();
                if (this.SongCount != null)
                    hashCode = hashCode * 59 + this.SongCount.GetHashCode();
                if (this.AlbumCount != null)
                    hashCode = hashCode * 59 + this.AlbumCount.GetHashCode();
                if (this.MusicVideoCount != null)
                    hashCode = hashCode * 59 + this.MusicVideoCount.GetHashCode();
                if (this.BoxSetCount != null)
                    hashCode = hashCode * 59 + this.BoxSetCount.GetHashCode();
                if (this.BookCount != null)
                    hashCode = hashCode * 59 + this.BookCount.GetHashCode();
                if (this.ItemCount != null)
                    hashCode = hashCode * 59 + this.ItemCount.GetHashCode();
                return hashCode;
            }
        }

    }
}
